lab-16-jacob/lib/authentication.js

```diff
@@ -0,0 +1,32 @@
+'use strict';
+
+let User = require('../model/user'); //basic auth middleware
+
+module.exports = (req, res, next) => {
+  let auth = req.headers.authorization;
+  if(!auth) {
+    res.status(400).end('needs auth');
+  }
+  let base64String = auth.split('Basic ')[1];
+  let [username, password] = new Buffer(base64String, 'base64').toString().split(':');
+  User.findOne({username: username})
+    .then(user => {
+      return user.comparePasswordHash(password)
+      .then((user) => {
```

Comment:
Try not to nest all this stuff. You can do all of this in your `.then()` chains from `findOne()`